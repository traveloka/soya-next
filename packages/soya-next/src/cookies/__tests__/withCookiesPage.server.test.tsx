/**
 * @jest-environment node
 */
import { Cookies } from "react-cookie";
import setupTest from "./utils/withCookiesPageSetup";

describe("withCookiesPage - Server", () => {
  it("should add cookie to page props", async () => {
    const { PageWithCookies } = setupTest();
    const props = await PageWithCookies.getInitialProps({
      req: {
        universalCookies: new Cookies("key=value"),
      },
    } as any);
    expect(props).toMatchSnapshot();
  });
});
