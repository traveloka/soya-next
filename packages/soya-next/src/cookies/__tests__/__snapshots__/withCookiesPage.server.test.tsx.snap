// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withCookiesPage - Server should add cookie to page props 1`] = `
Object {
  "cookies": Cookies {
    "HAS_DOCUMENT_COOKIE": false,
    "changeListeners": Array [],
    "cookies": Object {
      "key": "value",
    },
    "hooks": undefined,
  },
  "init": true,
}
`;
